File: Objetos_2_java/src/main/java/objetos2/java/gestionTareas/Usuario.java
package objetos2.java.gestionTareas;

import java.util.ArrayList;

public class Usuario {
    String nombre;
    ArrayList<Proyecto> proyectos;


    public Usuario(String nombre){
        this.nombre = nombre;
        this.proyectos = new ArrayList<>();
    }

    public Proyecto crearProyecto(String nombre){
        Proyecto proyecto = new Proyecto(nombre);

        if (!proyectos.contains(proyecto)){
            proyectos.add(proyecto);
            System.out.println("Proyecto '" + nombre + "' agregado con exito");
            return proyecto;
        }else{
            System.out.println("El proyecto '" + nombre + "' ya existe");
        }
        return null;
    }
}

File: Objetos_2_Kotlin/src/main/kotlin/gestorTareas/Usuario.kt
package gestorTareas

class Usuario(
    val nombre: String
) {

    private val proyectos = mutableListOf<Proyecto>()

    fun crearProyecto(nombre: String): Proyecto? {
        val proyecto = Proyecto(nombre)

        if (!proyectos.contains(proyecto)) {
            proyectos.add(proyecto)
            println("Proyecto '${proyecto.nombre}' ha sido creado")
            return proyecto
        }else{
            println("Este nombre de proyecto ya ha sido creado")
        }

        return null
    }
}